src/node/internal/internal_tls_wrap.ts

```diff
@@ -0,0 +1,644 @@
+  });
+};
+
+TLSSocket.prototype.renegotiate = function (
+  this: TLSSocket,
+  options: {
+    rejectUnauthorized?: boolean | undefined;
+    requestCert?: boolean | undefined;
+  },
+  callback?: (error: Error | null) => void
+): boolean {
+  validateObject(options, 'options');
+  if (callback !== undefined) {
+    validateFunction(callback, 'callback');
+  }
+  // TLS renegotiation is not supported.
+  return false;
+};
+
+TLSSocket.prototype.exportKeyingMaterial = function exportKeyingMaterial(
+  this: TLSSocket,
+  length: number,
+  label: string,
+  context?: Buffer
+): Buffer {
+  validateUint32(length, 'length', true);
+  validateString(label, 'label');
+  if (context !== undefined) {
+    validateBuffer(context, 'context');
+  }
+
+  throw new Error('exportKeyingMaterial is not implemented');
+};
+
+TLSSocket.prototype.setMaxSendFragment = function setMaxSendFragment(
+  this: TLSSocket,
+  size: number
+): boolean {
+  validateInt32(size, 'size');
+  // Setting maximum TLS fragment size is not supported.
+  return false;
+};
+
+TLSSocket.prototype._handleTimeout = function _handleTimeout(
+  this: TLSSocket
+): void {
+  this._emitTLSError(new ERR_TLS_HANDSHAKE_TIMEOUT());
+};
+
+TLSSocket.prototype._emitTLSError = function _emitTLSError(
+  this: TLSSocket,
+  err: Error
+): void {
+  const e = this._tlsError(err);
+  if (e) this.emit('error', e);
+};
+
+TLSSocket.prototype._tlsError = function _tlsError(
+  this: TLSSocket,
+  err: Error
+): Error | null {
+  this.emit('_tlsError', err);
+  if (this._controlReleased) return err;
+  return null;
+};
+
+TLSSocket.prototype._releaseControl = function _releaseControl(
+  this: TLSSocket
+): boolean {
+  if (this._controlReleased) return false;
+  this._controlReleased = true;
+  this.removeListener('error', this._tlsError.bind(this));
+  return true;
+};
+
+// This function is called from net.Socket onConnectionOpened() handler.
+TLSSocket.prototype._finishInit = function _finishInit(this: TLSSocket): void {
+  // Guard against getting onhandshakedone() after .destroy().
+  // * 1.2: If destroy() during onocspresponse(), then write of next handshake
+  // record fails, the handshake done info callbacks does not occur, and the
+  // socket closes.
+  // * 1.3: The OCSP response comes in the same record that finishes handshake,
+  // so even after .destroy(), the handshake done info callback occurs
+  // immediately after onocspresponse(). Ignore it.
+  if (!this._handle) return;
+
+  this._secureEstablished = true;
+  if (
+    this._tlsOptions.handshakeTimeout &&
+    this._tlsOptions.handshakeTimeout > 0
+  )
+    this.setTimeout(0, this._handleTimeout.bind(this));
+  this.emit('secure');
+};
+
+TLSSocket.prototype._start = function _start(this: TLSSocket): void {
+  if (this.connecting) {
+    this.once('connect', this._start.bind(this));
+    return;
+  }
+
+  // Socket was destroyed before the connection was established
+  if (!this._handle) return;
+
+  // this._handle.start();
```

Comment:
Why is this commented out?

Reply:
I was waiting for the correct test where we actually needed this, but it seems we can just make this function a noop for now.